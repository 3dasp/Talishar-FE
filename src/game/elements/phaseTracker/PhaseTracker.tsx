import React, { useEffect, useState } from 'react';
import styles from './PhaseTracker.module.css';
import PhaseTrackerWidget from '../phaseTrackerWidget/PhaseTrackerWidget';
import { useAppSelector } from '../../../app/Hooks';
import { RootState } from '../../../app/Store';

export default function PhaseTracker() {
  const turnPhase = useAppSelector((state: RootState) => state.game.turnPhase);
  const [phase, setPhase] = useState('');

  useEffect(() => {
    const newPhase = phaseTitle(turnPhase?.turnPhase);

    if (newPhase == null) {
      return;
    }
    setPhase(newPhase);
  }, [turnPhase]);

  return (
    <div className={styles.container}>
      <div className={styles.titleContainer}>
        <div className={styles.titleInterior}>{phase}</div>
      </div>
      <PhaseTrackerWidget />
      <div className={styles.phaseTrackerBottomContainer}>
        <div className={styles.bottomInterior}>{turnPhase?.caption}</div>
      </div>
    </div>
  );
}

function phaseTitle(turnPhase: string | undefined) {
  switch (turnPhase) {
    case 'M':
      return 'Main Phase';
    case 'B':
      return 'Defence Phase';
    case 'A':
      return 'Attack Reaction Step';
    case 'D':
      return 'Defence Reaction Step';
    case 'P':
      return 'Pitching';
    case 'ARS':
      return 'End Phase';
    case 'PDECK':
      return 'End Phase';
    default:
      return null;
  }
}
