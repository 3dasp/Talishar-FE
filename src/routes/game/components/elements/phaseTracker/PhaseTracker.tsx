import React, { useEffect, useState } from 'react';
import styles from './PhaseTracker.module.css';
import PhaseTrackerWidget from '../phaseTrackerWidget/PhaseTrackerWidget';
import { useAppSelector } from 'app/Hooks';
import { RootState } from 'app/Store';
import { shallowEqual } from 'react-redux';
import { setTokenSourceMapRange } from 'typescript';

export default function PhaseTracker() {
  const turnPhase = useAppSelector(
    (state: RootState) => state.game.turnPhase,
    shallowEqual
  );
  const [phase, setPhase] = useState('');
  const [phaseEnum, setPhaseEnum] = useState('');

  useEffect(() => {
    const newPhase = phaseTitle(turnPhase?.turnPhase);
    if (newPhase == null) {
      return;
    }
    if (turnPhase?.turnPhase != undefined) {
      setPhaseEnum(turnPhase.turnPhase);
    }
    setPhase(newPhase);
  }, [turnPhase]);

  return (
    <div className={styles.phaseTrackerContainer}>
      <div className={styles.titleContainer}>
        <div className={styles.titleInterior}>{phase}</div>
      </div>
      <PhaseTrackerWidget phase={phaseEnum} />
      <div className={styles.phaseTrackerBottomContainer}>
        <div
          className={styles.bottomInterior}
          dangerouslySetInnerHTML={{ __html: turnPhase?.caption ?? '' }}
        ></div>
      </div>
    </div>
  );
}

function phaseTitle(turnPhase: string | undefined) {
  switch (turnPhase) {
    case 'M':
      return 'Action Phase';
    case 'B':
      return 'Defend Step';
    case 'A':
      return 'Reaction Step';
    case 'D':
      return 'Reaction Step';
    case 'P':
      return 'Pitching';
    case 'ARS':
      return 'End Step';
    case 'PDECK':
      return 'End Step';
    default:
      return null;
  }
}
