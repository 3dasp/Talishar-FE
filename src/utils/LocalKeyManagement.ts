export const saveGameAuthKey = (gameId: number, authKey: string): void => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot save authKey for game:', gameId);
    return;
  }
  sessionStorage.setItem(String(gameId), authKey);
};

export const loadGameAuthKey = (gameId: number): string => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot load authKey for game:', gameId);
    return '';
  }

  const localValue = sessionStorage.getItem(String(gameId));
  if (!localValue) {
    console.error(
      'Attempted to access an authKey of a game that does not exist, game:' +
        gameId
    );
  }
  return localValue || '';
};

export const deleteGameAuthKey = (gameId: number): void => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot delete authKey for game:', gameId);
    return;
  }

  const valueExists = sessionStorage.getItem(String(gameId));
  if (!valueExists) {
    console.error(
      'Attempted to delete a authKey of a game that does not exist, game',
      gameId
    );
    return;
  }
  sessionStorage.removeItem(String(gameId));
};
