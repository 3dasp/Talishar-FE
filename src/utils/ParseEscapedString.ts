const CARDRE = /{{(.+?)\|(.+?)\|(.+?)}}/g;

export const replaceText = (inputString: string): string => {
  const mapping: { [key: string]: string } = {
    '0': 'gray',
    '1': 'red',
    '2': 'yellow',
    '3': 'blue'
  };

  return inputString.replace(CARDRE, (match, p1, p2, p3) => {
    const color = mapping[p3];
    const imgPath = `./WebpImages/${p1}.webp`;
    return `<span onmouseover="ShowDetail(event, '${imgPath}')" onmouseout="HideCardDetail()" style="color:${color}">${p2}</span>`;
  });
};
