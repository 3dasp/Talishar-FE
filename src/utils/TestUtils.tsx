/* eslint-disable @typescript-eslint/ban-types */
import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { AppStore, RootState, setupStore } from '../app/Store';

import InitialGameState from '../features/game/InitialGameState';
import { MemoryRouter } from 'react-router-dom';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    // @ts-ignore - I know I know...
    preloadedState = {
      game: InitialGameState
    },
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <MemoryRouter>{children}</MemoryRouter>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
