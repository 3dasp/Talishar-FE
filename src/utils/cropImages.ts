import { CLOUD_IMAGES_URL } from "appConstants";

export const generateCroppedImageUrl = (cardName: string): string => `${CLOUD_IMAGES_URL}/crops/${cardName}_cropped.png`;