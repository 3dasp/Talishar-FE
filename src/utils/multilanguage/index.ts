export {
  getCollectionCardImagePath,
  getCardBackImagePath,
  loadInitialLanguage
} from './multilanguage';
export type { pathType } from './types';
export {
  DEFAULT_LANGUAGE,
  CARD_IMAGES_PATH,
  CARD_SQUARES_PATH
} from './constants';
