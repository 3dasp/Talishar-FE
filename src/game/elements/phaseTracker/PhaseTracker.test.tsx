import React from 'react';
import PhaseTracker from './PhaseTracker';
import { renderWithProviders } from '../../../utils/TestUtils';

it('renders without crashing', () => {
  const div = renderWithProviders(<PhaseTracker />);
  const display = document.querySelector('div');
  expect(display).toMatchSnapshot();
});
