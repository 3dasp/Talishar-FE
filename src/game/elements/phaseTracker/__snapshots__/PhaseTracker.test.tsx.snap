// Vitest Snapshot v1

exports[`renders without crashing 1`] = `
<div>
  <div
    class="_container_eeb0ac"
  >
    <div
      class="_titleContainer_eeb0ac"
    >
      <div
        class="_titleInterior_eeb0ac"
      />
    </div>
    <div
      class="_phaseWidget_58c017"
    >
      <div
        class="_icon_58c017"
        style="left: 0em;"
      />
    </div>
    <div
      class="_phaseTrackerBottomContainer_eeb0ac"
    >
      <div
        class="_bottomInterior_eeb0ac"
      />
    </div>
  </div>
</div>
`;
